squad/core/history.py

```diff
@@ -77,14 +77,25 @@ def __init__(self, project, full_test_name, top=None, page=1, per_page=20):
         suite = project.suites.prefetch_related('metadata').get(slug=suite_slug)
         metadata = SuiteMetadata.objects.get(kind='test', suite=suite_slug, name=test_name)
         tests = self.__get_tests__(builds, metadata)
+        all_envs = set(project.environments.all())
         for test in tests:
             build = test.test_run.build
             environment = test.test_run.environment
-
             environments[environment] = True
             known_issues = issues_by_env.get(environment.id)
 
             results[build][environment] = TestResult(test, suite, metadata, known_issues)
+            remaining_envs = all_envs - set([environment])
+            for env in remaining_envs:
+                results[build][env] = None
+                environments[env] = True
+
+        # Make sure all builds that don't have the test have None at least
+        for b in builds:
+            if not results[build]:
```

Comment:
hmm, this is weird, loop variable is 'b' not 'build'. Is this a typo?

Reply:
@mwasilew ouch..yes..typo. thanks for the catch